docs/sphinx/data_center_install.rst

```diff
@@ -268,13 +273,98 @@ You can then create a self-extracting archive with the command
         "CUDA Quantum toolkit for heterogeneous quantum-classical workflows" \
         bash cudaq/migrate_assets.sh -t /opt/nvidia/cudaq
 
+.. _cudaq-python-from-source:
+
+Python Support
++++++++++++++++++++++++++++++++
+
+The most convenient way to enable Python support within CUDA Quantum is to build 
+a `wheel <https://pythonwheels.com/>`__ that can then easily be installed
+using `pip`. To ensure the wheel can be installed on the host system, make sure to
+use the same Python version for the build as the one that is installed on the host system.
+To build the CUDA Quantum Python wheel, you will need to install the following additional 
+Python-specific tools:
+
+- Python development headers: The development headers for your Python version are installed
+  in the way as you installed Python itself. If you installed Python via the package manager
+  for your system, you may need to install an additional package to get the development headers.
+  The package name is usually your python version followed by either a `-dev` or `-devel` suffix.
+  If you are using a `Conda environment <https://conda.io/projects/conda/en/latest/user-guide/getting-started.html#managing-python>`__,
+  the necessary headers should already be installed.
```

Comment:
Why? It is not necessary for the build.
Are you suggesting to install it for debugging purposes when developing CUDA Quantum? The purpose of this document is to give a clear, simple and tested way to build and install CUDA Quantum for using it without installing anything else. This is not the same guidance as setting up a dev environment to develop CUDA Quantum itself - that should only be documented on the repo, not in the public CUDA Quantum docs. I realize that I updated the repo guidance to point to this page for building from scratch without using a dev container, so maybe that link should be removed if it causes confusion.

Reply:
It wasn't clear that the audience was people that only wanted to build the toolchain in "release mode" to me.

Why wouldn't we want to add documentation for a wider audience that includes people that may want to contribute to the project?

Reply:
Not saying it wouldn't be good to add docs for people that want to contribute, but I think we would need a separate guide for these. The requirements for local development + contributing are simply pretty different from the requirements for build + deploy in production. E.g. for local development, what you say about starting with a minimal setup makes sense, whereas for deploy and build in production the guide really needs to produce a fully featured installation with all the bells and whistles. Also, local development means we can easily recommend/set up codespaces, which is precisely facilitating this scenario across GitHub, whereas for production deployment, build in one environment and run in a different environment is very important. 
We do already have separate guidance for developers, which is on the GitHub repositories, and we can/should more details there. This page is linked specifically for using CUDA Quantum on systems that we don't distribute any pre-built packages/images for, or e.g. for organizations that have other constraints (such as e.g. specific CUDA versions, or a requirement to use only what they built themselves).

Reply:
I'd urge you to consider having a single set of build instructions instead of a set of distinct build instructions for different contexts. This will make the documentation easier to maintain, and in my opinion better overall.

We can add tabs to the pages for different sets of options and add-on features that are useful under different scenarios, much like the CUDA Quantum Spec has C++ and Python tabs.

Again, there is a different between "how we did it in our scripts" vs. "the bare minimum". For example, with the `lld` discussion: I almost never build our cuda-quantum with `lld`, and I fully expect everything to continue to work seamlessly.